pyscripts/build_docker_image.py

```diff
@@ -10,193 +10,237 @@
-                    image_tag,
-                    stream=True,
-                    auth_config=auth_config,
+            for line in api_client.build(
+                path=get_dockerfile_directory(tags),
+                tag=tags.docker_image_name,
+                buildargs=build_args,
+                cache_from=cache_from,
+                dockerfile=dockerfile,
+            ):
+                print_filtered_docker_line(line)
+
+        def tag_images(tags):
+            print(print_separator)
+            print("TAGGING IMAGE")
+            print(print_separator)
+
+            if sys.platform == 'win32':
+                base_url = "tcp://127.0.0.1:2375"
+            else:
+                base_url = "unix://var/run/docker.sock"
+            api_client = docker.APIClient(base_url=base_url)
+
+            print("Adding tags")
+            for image_tag in tags.image_tags:
+                print("Adding " + image_tag)
+                api_client.tag(tags.docker_image_name, tags.docker_full_image_name, image_tag.lower())
+
+
+        def push_images(tags):
+            print(print_separator)
+            print("PUSHING IMAGE")
+            print(print_separator)
+
+            if sys.platform == 'win32':
+                base_url = "tcp://127.0.0.1:2375"
+            else:
+                base_url = "unix://var/run/docker.sock"
+            api_client = docker.APIClient(base_url=base_url)
+
+            for image_tag in tags.image_tags:
+                print("Pushing {}:{}".format(tags.docker_full_image_name, image_tag))
+                for line in api_client.push(
+                    tags.docker_full_image_name, image_tag.lower(), stream=True, auth_config=auth_config
                 ):
-                    print_filtered_docker_line(line)
-                tags.image_tag_to_use_for_cache = image_tag
-                print(
-                    Fore.GREEN
-                    + "Found {}.  Using this for image cache".format(image_tag)
-                )
-                return
-            except docker.errors.APIError:
-                print(Fore.YELLOW + "Image not found in repository")
-
-
-tags = docker_tags.get_docker_tags_from_commit(
-    args.language, args.repo, args.commit, args.variant
-)
-
-prefetch_cached_images(tags)
-build_image(tags)
-tag_images(tags)
-push_images(tags)
-
-if not docker_tags.running_on_azure_pipelines():
-    print(Fore.GREEN + "Done.  Deploy with the following command:")
-    print(
-        Fore.GREEN
-        + "./deploy-test-containers.sh --{} {}:{}".format(
-            tags.language, tags.docker_full_image_name, tags.image_tags[0]
+                    try:
+                        sys.stdout.write(json.loads(line.decode("utf-8"))["stream"])
+                    except:
+                        try:
+                            print_filtered_docker_line(line)
+                        except:
+                            out_line = ''.format("{}",line)
```

Comment:
>out_line = ''.format("{}",line) [](start = 28, length = 31)

Is this some sort of magic or a typo?  I _think_ you mean "{}".format(line), or do you mean something else?  also, why not just use str(line) (here and everywhere)?

Reply:
fixed with line.decode()